main/admin/init_cas_user_extra_field_from_ldap.php

```diff
@@ -0,0 +1,323 @@
+    if (false === $searchResult) die('ldap_search() failed: '.ldap_error($ldap)."\n");
+    $userId = $user->getId();
+    print "$username ($userId): ";
+    switch (ldap_count_entries($ldap, $searchResult)) {
+        case 0:
+            print "does not exist in the LDAP directory, skipping.\n";
+            break;
+        case 1:
+            $entry = ldap_first_entry($ldap, $searchResult);
+            if (false === $entry) die('ldap_first_entry() failed: '.ldap_error($ldap)."\n");
+            $ldapCASUser = ldap_get_values($ldap, $entry, $ldapCASUserAttribute)[0];
+            if (false === $ldapCASUser) die('cannot read CAS user code from LDAP entry: '.ldap_error($ldap)."\n");
+            $ldapUsername = ldap_get_values($ldap, $entry, $ldapUsernameAttribute)[0];
+            if (false === $ldapUsername) die('cannot read username from LDAP entry: '.ldap_error($ldap)."\n");
+            print "\033[2K\r$ldapUsernameAttribute: $ldapUsername, $ldapCASUserAttribute: $ldapCASUser, ";
+            $problems = [];
+            if ($username === $ldapUsername) {
+                // fine
+            } else if (
+                strtolower(trim($username)) === strtolower(trim($ldapUsername))
+                ||
+                strtolower(trim($username)) === strtolower(trim($ldapCASUser))
+            ) {
+                if (array_key_exists($ldapUsername, $userNamesInUse)) {
+                    print "wrong username but '$ldapUsername' is already taken, skipping.\n";
+                    break;
+                } else {
+                    $problems[] = "wrong username";
+                    $wrongUserNames[$userId] = $ldapUsername;
+                    $userNamesInUse[$ldapUsername] = $userId;
+                }
+            } else {
+                die("LDAP search result does not match username; our filter is wrong: $filter\n");
+            }
+            if (array_key_exists($userId, $existingCasUserValues)) {
+                $currentValue = $existingCasUserValues[$userId];
+                if ($currentValue !== $ldapCASUser) {
+                    $problems[] = "wrong current CAS user code '$currentValue'";
+                    $wrongCASCodes[$userId] = $ldapCASUser;
+                }
+            } else {
+                $problems[] = "CAS user code missing in database";
+                $missingCASCodes[$userId] = $ldapCASUser;
+            }
+            $currentAuthSource = $user->getAuthSource();
+            if (CAS_AUTH_SOURCE === $currentAuthSource) {
+            } else {
+                $problems[] = "wrong auth source '$currentAuthSource'";
+                $wrongAuthSources[$userId] = true;
+            }
+            print (empty($problems) ? "ok\r" : (join(', ', $problems)."\n"));
+            break;
+        default:
+            print "more than one entries for username '$username' in the LDAP directory for user id=$userId, skipping.\n";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line exceeds maximum limit of 120 characters; contains 122 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=4906847)